submissions/AndriiZahumnyi/js-dom/script/index.js

```diff
@@ -0,0 +1,39 @@
+
+const hamburgerToggle = document.querySelector('#menu-toggle');
+
+const flexMenu = document.querySelector('.flex-menu');
+
+hamburgerToggle.addEventListener('click', function (e) {
+
+  flexMenu.classList.toggle('active');
+
+});
+
+const arrAside = [".aside-html", ".aside-css", ".aside-bootstrap", ".aside-js", ".aside-jquery", ".aside-git"];
+const arrContent = [".html-section", ".css-section", ".bootstrap-section", ".js-section", ".jquery-section", ".git-section"];
+
+
+const asideMenuList = document.getElementById("aside-menu-list");
+
+const n = arrAside.length;
+
+function respondToTheClick(evt) {
+
+  for (let i = 0; i < n; i++) {
```

Comment:
I'd recommend using `Array.prototype.find` here instead of `for` loop. It can help you to decrease the number of operations. Let's imagine that user clicked to the second element in the menu. In this case you should iterate through ALL elements and check the content.
Using the `Array.prototype.find` method will help you to stop the loop after fulfilling the condition

Reply:
It seems to me that Array.prototype.find can`t help here, because if the last elements was chosen and we need first or second, we have to set:

 sectionContent.classList.add("section-none");
 sectionContent.classList.remove("section-block");

for that last one. Otherwise it just stay on the screen.
So, full loop through have to be. 
What do you say?

Reply:
> I'd recommend using `Array.prototype.find` here instead of `for` loop. It can help you to decrease the number of operations. Let's imagine that user clicked to the second element in the menu. In this case you should iterate through ALL elements and check the content.
> Using the `Array.prototype.find` method will help you to stop the loop after fulfilling the condition

i have tried to use `Array.prototype.find`, but i can`t come up with deactivate previous looks of content...

function respondToTheClick(evt) {

  const asideButton = arrAside.find(element => document.querySelector(element) === evt.target);
  const indexOfAsideButton = arrAside.indexOf(asideButton);
  const sectionContent = document.querySelector(arrContent[indexOfAsideButton]);
  console.log(sectionContent);

  sectionContent.classList.add("section-block");
  sectionContent.classList.remove("section-none");

  /* for (let i = 0; i < lengthOfArrAside; i++) {
    
    const asideButton = document.querySelector(arrAside[i]);
    const sectionContent = document.querySelector(arrContent[i]);

    if (asideButton !== evt.target) {
      sectionContent.classList.add("section-none");
      sectionContent.classList.remove("section-block");
    } else {
      sectionContent.classList.add("section-block");
      sectionContent.classList.remove("section-none");
    }

  } */

}